components/pages/workPageLayout/WorkPageHeader.tsx

```diff
@@ -1,51 +1,54 @@
+import { useRouter } from "next/navigation"
+import React from "react"
 
 import WorkPageButtons from "@/components/pages/workPageLayout/WorkPageButtons"
 import {
-  addMaterialTypeIconToSelectOption,
-  findInitialSliderValue,
-  getManifestationByMaterialType,
   getManifestationLanguageIsoCode,
   getWorkMaterialTypes,
-  translateMaterialTypesForRender,
+  materialTypeSortPriority,
+  translateMaterialTypesStringForRender,
 } from "@/components/pages/workPageLayout/helper"
 import { Badge } from "@/components/shared/badge/Badge"
 import { CoverPicture } from "@/components/shared/coverPicture/CoverPicture"
 import SlideSelect, { SlideSelectOption } from "@/components/shared/slideSelect/SlideSelect"
 import {
   GeneralMaterialTypeCodeEnum,
+  ManifestationWorkPageFragment,
+  Work,
   WorkFullWorkPageFragment,
 } from "@/lib/graphql/generated/fbi/graphql"
 import { getCoverUrls, getLowResCoverUrl } from "@/lib/helpers/helper.covers"
 import { displayCreators } from "@/lib/helpers/helper.creators"
+import { resolveUrl } from "@/lib/helpers/helper.routes"
 import { getIsbnsFromManifestation } from "@/lib/helpers/ids"
 import { useGetCoverCollection } from "@/lib/rest/cover-service-api/generated/cover-service"
 import { GetCoverCollectionSizesItem } from "@/lib/rest/cover-service-api/generated/model"
 import { useGetV1ProductsIdentifier } from "@/lib/rest/publizon-api/generated/publizon"
-import { useSelectedManifestationStore } from "@/store/selectedManifestation.store"
 
 type WorkPageHeaderProps = {
   work: WorkFullWorkPageFragment
+  selectedManifestation: ManifestationWorkPageFragment
 }
 
-const WorkPageHeader = ({ work }: WorkPageHeaderProps) => {
-  const searchParams = useSearchParams()
+const WorkPageHeader = ({ work, selectedManifestation }: WorkPageHeaderProps) => {
   const router = useRouter()
-  const { selectedManifestation, setSelectedManifestation } = useSelectedManifestationStore()
-  const [slideSelectOptions, setSlideSelectOptions] = useState<SlideSelectOption[] | null>(null)
   const isbns = selectedManifestation ? getIsbnsFromManifestation(selectedManifestation) : []
   const languageIsoCode = getManifestationLanguageIsoCode(selectedManifestation)
   const titleSuffix = selectedManifestation?.titles?.identifyingAddition || ""
-  const [initialSliderValue, setInitialSliderValue] = useState<SlideSelectOption | undefined>(
-    undefined
+  const workMaterialTypes = getWorkMaterialTypes(
+    (work?.materialTypes as Work["materialTypes"]) || []
   )
-  const workMaterialTypes = getWorkMaterialTypes(work.materialTypes).map(materialType => {
-    return { value: materialType.code, render: materialType.display }
-  })
+  const workMaterialTypesWithDisplayName = workMaterialTypes.map(materialType => {
+    return {
+      code: materialType.code,
+      display: translateMaterialTypesStringForRender(
+        materialType.code as GeneralMaterialTypeCodeEnum
+      ),
+    }
+  }) as SlideSelectOption[]
```

Comment:
This is something to consider:
When you write a function inside the component (which essentially is another function) it will be **defined** every time the component is called (which can be quite a lot of times). It doesn't necessarily it will have an impact, but it can have an impact both in computation and in memory use.
Because of that I think it is a good practise, at least to consider writing the function outside of the component to avoid such pitfalls. Again it depends on the case but if the argument is: but this is easier, I don't care about the functions being pure it is definitely the wrong approach. Another benefit of having isolated functions that you can test them isolated i unit tests. That is not possible if they are defined inside the components.

Reply:
This also looks like a function that would be nice to have globally. But I could be wrong.